tests/mod.rs

```diff
@@ -644,3 +644,94 @@ fn no_warning_for_deprecated() {
 		VariantB,
 	}
 }
+
```

Comment:
```suggestion
#[test]
fn decoding_array_does_not_stack_overflow() {
	let data = &[];
	let _ = <[u8; 100 * 1024 * 1024]>::decode(&mut data.as_slice());
}
```
This one still fails…

Reply:
Can we check the maximal allocateable stack size in a non-hacky way and bail early?  
Not sure what really to do either. Just wanted to point out that this could still panic…

Reply:
> Can we check the maximal allocateable stack size in a non-hacky way and bail early?

Hm... using Linux-specific APIs - possibly. We could grab the stack base, grab the stack pointer, and grab the maximum stack size, and see how much's left.

Not sure if we actually want to do that; seems a little bit excessive. (And it won't work inside e.g. WASM)

> Do you know how do other decoding libs handle this?

AFAIK, they don't. (:

Reply:
I looked at `serde` and they only support arrays up to 32 entries.  

Anyway, can you maybe add a "known issues" or something so that we can link it if people report it as bug again in the future? And maybe a "should_panic" test to verify that arrays are not covered.

Reply:
I can add a note somewhere, but as far as a test is concerned that can't be done with `should_panic` as a stack overflow doesn't actually produce a panic. (We could maybe test it by spawning a new process, but I'm not convinced it's actually worth it.)

Reply:
Yes that sounds good, thanks!